autoupdate.js

```diff
@@ -0,0 +1,57 @@
+const autoUpdater = require('electron-updater').autoUpdater
+const { dialog } = require('electron');
+
+const hour = 60 * 60;
+const autoUpdaterInterval = hour * 1000;
+
+const RESTART_BUTTON = 0;
+const LATER_BUTTON = 1;
+
+function autoUpdateDisabled(config) {
+  return process.mas || config.get('disableAutoUpdate');
+}
+
+function checkForUpdates() {
+  autoUpdater.checkForUpdates();
+}
+
+function showUpdateDialog() {
+  const options = {
+    type: 'info',
+    buttons: ['Restart', 'Later'],
+    title: "Signal update available",
+    message: "There is a new version of Signal available.",
+    detail: "Press Restart to apply the updates.",
+    defaultId: RESTART_BUTTON,
+    cancelId: LATER_BUTTON
+  }
+
+  dialog.showMessageBox(options, function(response) {
+    if (response == RESTART_BUTTON) {
+      autoUpdater.quitAndInstall();
+    }
+  });
+}
+
+function onUpdateAvailable() {
+  showUpdateDialog();
+}
+
+function onError(error) {
+  console.log("Got an error while updating: ", error);
+}
+
+function initializeAutoUpdater(config) {
+  if (autoUpdateDisabled(config)) { return; }
+
+  autoUpdater.addListener('update-available', onUpdateAvailable);
```

Comment:
Shouldn't this be 'update-downloaded'? Otherwise I believe the user could quit and restart before the download completes... By default, the updater downloads the update as soon as it's available, but it's a 50MB+ download which could take some time.